### PR TITLE
impl ldap search template

### PR DESCRIPTION
**Pull request by yevheniimuzyka**

### CODE REVIEW

File: src/main/java/com/icthh/xm/uaa/web/rest/LdapResource.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+@RequestMapping("/api")
+@AllArgsConstructor
+public class LdapResource {
+
+    private final LdapService ldapService;
+
+    @GetMapping("/ldap/_search-with-template")
+    @Timed
+    @PreAuthorize("hasPermission(null, 'LDAP.SEARCH')")
+    public ResponseEntity searchByTemplate(@NotEmpty @RequestParam String templateKey,
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/uaa/web/rest/LdapResource.java]: add #templateKey as parameter of hasPermission

File: src/main/java/com/icthh/xm/uaa/service/LdapService.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    private TenantProperties.Ldap foundLdap(String ldapDomain) {
+        List<TenantProperties.Ldap> ldapProperties = tenantPropertiesService.getTenantProps().getLdap();
+
+        return ldapProperties
+            .stream()
+            .filter(ldap -> ldapDomain.equals(ldap.getDomain()))
+            .findFirst()
+            .orElseThrow(() -> new RuntimeException("Ldap with domain: " + ldapDomain + " not found"));
```
Comment (major13ua) [inline src/main/java/com/icthh/xm/uaa/service/LdapService.java]: Please use specific exception inherited from RuntimeException to intercept and process response accordingly

File: src/main/java/com/icthh/xm/uaa/service/LdapService.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+        List<TenantProperties.LdapSearchTemplate> searchTemplates = tenantPropertiesService.getTenantProps()
+                                                                                           .getLdapSearchTemplates();
+
+        TenantProperties.LdapSearchTemplate searchTemplate =
+            searchTemplates
+            .stream()
+            .filter(template -> templateKey.equals(template.getTemplateKey()))
+            .findFirst()
+            .orElseThrow(() -> new RuntimeException("Search template with templateKey: " + templateKey + " not found"));
```
Comment (major13ua) [inline src/main/java/com/icthh/xm/uaa/service/LdapService.java]: Please use specific exception inherited from RuntimeException to intercept and process response accordingly

File: src/main/java/com/icthh/xm/uaa/service/LdapService.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+public class LdapService {
+
+    private TenantPropertiesService tenantPropertiesService;
+
+    @LogicExtensionPoint(value = "SearchByTemplate", resolver = LdapSearchByTemplateKeyResolver.class)
+    @Transactional(readOnly = true)
+    public Set<Map<String, List<String>>> searchByTemplate(String templateKey,
+                                                           TemplateParams templateParams) {
+
+        Object[] params = templateParams.getTemplateParams().toArray();
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/uaa/service/LdapService.java]: Potential NPE if there are no `templateParams`. Pls, also add test, when `templateParams` not needed.

**Pull request merged**